### DISCUSSION TITLE
How to send nuclei outrput to a database to keep historical scans?

### DISCUSSION THREAD

**Discussion by MarsIsRed** [projectdiscovery/nuclei#3336] [Q&A] [answered]
I see Splunk and other options but can it be sent to other databases.

**Comment by forgedhallpass** [comment, marked as answer]
Everything that is currently supported is documented under [nuclei reporting](https://nuclei.projectdiscovery.io/nuclei/get-started/#nuclei-reporting) in the documentation.

Generic support for writing the output into custom databases is not practical because it would require externalizing a lot of configurations (e.g. the table name to insert into, column names, mappings etc.)

If by "databases" you meant services that could be leveraged by other users as well, you can create a *Feature Request* under the [Issues](https://github.com/projectdiscovery/nuclei/issues/new/choose) section. You are also more than welcome to open a PR with the implementation.

[Here](https://github.com/projectdiscovery/nuclei/blob/25fcae14930320117fc742c1a61d93b53b6ef67d/v2/pkg/reporting/reporting.go#L124) you can find the relevant part of the code that can be extended.

**Reply by mrngm** [reply]
For another project, I was looking into the ElasticSearch reporter to see how the reporting functionality worked. It seems easy enough to implement a feature that lets nuclei report to a more generic HTTP endpoint with the [`output.ResultEvent` struct](https://github.com/projectdiscovery/nuclei/blob/7ddbbf8e4db28a34b3a8e4dd1452692f4bb279c9/v2/pkg/output/output.go#L79-L127) marshaled into JSON.

I could shoot in a PR for this; it seems simply implementing the [`reporting.Exporter`](https://github.com/projectdiscovery/nuclei/blob/7ddbbf8e4db28a34b3a8e4dd1452692f4bb279c9/v2/pkg/reporting/reporting.go#L78-L84) interface suffices?
